runtime/hash.c

```diff
@@ -205,14 +205,12 @@ CAMLprim value caml_hash(value count, value limit, value seed, value obj)
       h = caml_hash_mix_intnat(h, v);
       num--;
     }
-#ifndef NO_NAKED_POINTERS
```

Comment:
This one is temporarily reintroducing a page-table check into the NNP runtime. You then remove it in commit 8c062185cc422d759d7084afcb732d1b9bec5bf8.
I suggest that you move that commit before this one, and change the commit message (of 8c062185cc422d759d7084afcb732d1b9bec5bf8) because it wrongly claims that there is only one use of `Is_in_value_area` in `finalise.c`.

Reply:
Sorry I merged during your review, didn't know you were on it.  

At some point Github was displayign the commits in the wrong order.

I think we're OK here: first 6e9d39bb2 defines `Is_in_value_area` to 1 in no-naked-pointers mode, then 247c1da51 removes the `#ifdef` above.

>  it wrongly claims that there is only one use of Is_in_value_area in finalise.c

I'm not seeing this claim in the commit messages.

Reply:
And now you can turn your attention to #9704 :-)

Reply:
> it wrongly claims that there is only one use of Is_in_value_area in finalise.c.

I see you meant Is_in_heap_or_young.  And I meant only one use outside assertions.